liferay-uniform/liferay-portal
# LPS-197686 Have a more accurate upgrade result

From https://github.com/liferay-uniform/liferay-portal/pull/1597

Issue:
The upgrade status of an upgrade is currently influenced by the presence of any error.
This causes the upgrade to complete with a status that may not be accurate.

Solution:
The aim is to calculate the Upgrade Status based on the combine result of:

Core Upgrades
Module Upgrades
Verify Process Exceptions
Unsatisfied Components

The upgrade will only have one of three statuses upon completion:

Success - Needs 1 & 2 & 3 & 4 to Pass
Failure - Needs 1 OR 2 OR 3 to Fail
Unresolved - Needs 4 to Fail

Based on this, we can ignore errors that occur during the upgrade process that may or may not be relevant.

Ticket:
[LPS-197686](https://liferay.atlassian.net/browse/LPS-197686)

Tests were modified to account for the changes.
Tests were added to account for all of the different statuses and cause of the status.

---

QA Notes:

- **CheckUpgradeClientUpgradeCoreOnly** is updated to expect `failure` upgrade status since a core module has failed. https://github.com/liferay-uniform/liferay-portal/commit/c55e969798276940d018ecc80e560f5bb9a2185a
- **VerifyLayoutFriendlyURLErrorOnlyInUpgradeProcess** is updated to no longer connect to the Gogoshell after upgrade because it throws a warning during upgrade (an irrelevant issue because the upgrade is successful). https://github.com/liferay-uniform/liferay-portal/commit/583a98ad116c0e0840ba58dd8b42171e9d84febf
- **CheckMbeanWhenUpgradeItsNotExecuted** is updated to expect `major` upgrade type even if no upgrade processes are executed because the entries for the modules don't exist yet. https://github.com/liferay-uniform/liferay-portal/commit/00fccd67d4e52f99fd6a338f1172485966ad9316
- **CheckMbeanWarningUpgrade**, **WarningsLogged** is updated to expect `success` upgrade status because the warning status is deprecated. Warnings still show up in the upgrade report. https://github.com/liferay-uniform/liferay-portal/commit/b7b2628ec3ea48b6a13ea860d0ffbd5f79607423

:warning:  **NOTE:** DO NOT FORWARD this PR. I temp reverted https://github.com/liferay-uniform/liferay-portal/commit/44c3af8aefdb267e552e780b6bc22ca7999ce694 to get better upgrade results. I will manually forward if it passes review.
